src/sendalert.cpp

```diff
@@ -0,0 +1,201 @@
+// Original code from: https://gist.github.com/laanwj/0e689cfa37b52bcbbb44
+
+/*
+
+To set up a new alert system
+----------------------------
+
+Create a new alert key pair:
+openssl ecparam -name secp256k1 -genkey -param_enc explicit -outform PEM -out data.pem
+
+Get the private key in hex:
+openssl ec -in data.pem -outform DER | tail -c 279 | xxd -p -c 279
+
+Get the public key in hex:
+openssl ec -in data.pem -pubout -outform DER | tail -c 65 | xxd -p -c 65
+
+Update the public keys found in chainparams.cpp.
+
+
+To send an alert message
+------------------------
+
+Copy the private keys into alertkeys.h.
+
+Modify the alert parameters and message found in this file.
+
+Build and run to send the alert (after 60 seconds):
+
+./zcashd -printtoconsole -sendalert
+
+*/
+
+/*
+So you need to broadcast an alert...
+... here's what to do:
+
+1. Copy sendalert.cpp into your bitcoind build directory
+
+2. Decrypt the alert keys
+  copy the decrypted file as alertkeys.h into the src/ directory.
+
+3. Modify the alert parameters in sendalert.cpp
+  See the comments in the code for what does what.
+
+4. Add sendalert.cpp to the src/Makefile.am so it gets built:
+
+    libbitcoin_server_a_SOURCES = \
+      sendalert.cpp \
+      ... etc
+
+5. Update init.cpp to launch the send alert thread. 
+  Define the thread function as external at the top of init.cpp:
+
+    extern void ThreadSendAlert();
+
+  Add this call at the end of AppInit2:
+
+    threadGroup.create_thread(boost::bind(ThreadSendAlert));
+
+6. build bitcoind, then run it with -printalert or -sendalert
+  I usually run it like this:
+   ./bitcoind -printtoconsole -sendalert
+
+One minute after starting up the alert will be broadcast. It is then
+flooded through the network until the nRelayUntil time, and will be
+active until nExpiration OR the alert is cancelled.
+
+If you screw up something, send another alert with nCancel set to cancel
+the bad alert.
+*/
+#include "main.h"
+#include "net.h"
+#include "alert.h"
+#include "init.h"
+
+#include "util.h"
+#include "utiltime.h"
+#include "key.h"
+#include "clientversion.h"
+#include "chainparams.h"
+
+#include "alertkeys.h"
+
+
+static const int64_t DAYS = 24 * 60 * 60;
+
+void ThreadSendAlert()
+{
+    if (!mapArgs.count("-sendalert") && !mapArgs.count("-printalert"))
+        return;
+
+    MilliSleep(60*1000); // Wait a minute so we get connected
+
+    //
+    // Alerts are relayed around the network until nRelayUntil, flood
+    // filling to every node.
+    // After the relay time is past, new nodes are told about alerts
+    // when they connect to peers, until either nExpiration or
+    // the alert is cancelled by a newer alert.
+    // Nodes never save alerts to disk, they are in-memory-only.
+    //
+    CAlert alert;
+    alert.nRelayUntil   = GetTime() + 15 * 60;
+    alert.nExpiration   = GetTime() + 365 * 60 * 60;
+    alert.nID           = 1040;  // use https://en.bitcoin.it/wiki/Alerts to keep track of alert IDs
```

Comment:
Ok, now set to 1000 and URL points to z.cash.  Can update with specific page when it exists.